Added spin parsing to raw_parser

asamli opened the pull request:
Added parsing for spins from the Mulliken analysis section

asamli committed:
Added spin parsing to raw_parser

Added parsing for spins from the Mulliken analysis section

zhubonan reviewed (commented):
Thanks, can you please add a test as well? You can resuse some existing test data. 

Also I think it is better to put the functionality into a function for reading the population block in general (similar to `read_force_box`). 
Beear in mind that, not every calculation is spin polarised so we need to be sure it is read only when it is appropriate.

asamli committed:
Merge branch 'zhubonan:dev' into dev

asamli committed:
Updated Mulliken parser and added test

-Updated the Mulliken population box parser to be a function which parses charges as well as spins
-Added a test for the population box parser

asamli committed:
Fixed Mulliken parser

Removed leftover code and fixed the population parser

asamli committed:
Desorting charges and spins from Mulliken parser

Desorted the charges and spins from the Mulliken parser in raw_parser.py

zhubonan commented:
Merged, thanks!
